include/irrArray.h

```diff
@@ -549,11 +363,11 @@ class array
 	is returned. */
 	s32 linear_reverse_search(const T& element) const
 	{
-		for (s32 i=used-1; i>=0; --i)
-			if (data[i] == element)
-				return i;
-
-		return -1;
+		auto it = std::find(m_data.rbegin(), m_data.rend(), element);
+		if (it == m_data.rend())
+			return -1;
+		size_t offset = it - m_data.rbegin();
+		return m_data.size() - offset - 1;

```

Comment:
Feel free to add basic unit tests [here](https://github.com/minetest/irrlicht/blob/master/examples/AutomatedTest/main.cpp), I can't tell at a glance whether this is correct.

Reply:
Added unit tests in another PR:

https://github.com/minetest/irrlicht/pull/103

Both the existing code and the new version pass.